pkg/plugin/system/system.go

```diff
@@ -105,12 +112,61 @@ func (ps *pluginSystem) Configure(streams ioutils.Streams) error {
 		}
 
 		aspectplugin := rawplugin.(plugin.Plugin)
-		ps.plugins.insert(aspectplugin)
+
+		node := &PluginNode{
```

Comment:
Why do we create a new `PluginNode` type instead of putting the custom commands into the `aspectplugin`?

I'm assuming `aspectplugin := rawplugin.(plugin.Plugin)` converts the public API into an internal representation? Should that also wrap the custom commands?

Reply:
`aspectplugin := rawplugin.(plugin.Plugin)` is casting `rawplugin` to `plugin.Plugin` which is public facking and I dont think we want those fields on there.